### PR TITLE
Add more CI checks

### LINKED ISSUE #744: Add dependency security and license checks
**Is your feature request related to a problem? Please describe.**
Currently, we do not check dependencies for known security vulnerabilities and licenses automatically.

**Describe the solution you'd like**
Add security checks (https://pyup.io/safety/) and license checks (https://github.com/dhatim/python-license-check) for dependencies

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Add `safety`, `liccheck` and `copyright_check` checks.

Summary of changes:
- Introduced `safety` testenv, and added it to Jenkinsfile.
- Introduced `liccheck` test, and added it to Jenkinsfile.
- Added script `scripts/parse_main_dependencies_from_lock.py`. The purpose is to extract the main (non-development) dependencies in `requirements.txt` format, so it can be parsed by `liccheck`.
- Introduced script `scripts/check_copyright_notice.py`, that checks the consistency of all the Python files header comments.
- Added testenv `copyright_check`, which runs the aforementioned script.
- Updated headers in Python files to make the `copyright_check` to pass.
- Fixed an issue with the testenv `docs`

## Fixes

Fixes #744 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [X] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [X] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [X] Lint and unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked that code coverage does not decrease.
- [X] I have checked that the documentation about the `aea cli` tool works
- [X] I have added necessary documentation (if appropriate)
- [X] Any dependent changes have been merged and published in downstream modules

## Further comments

Is `copyright_check` a good name? Maybe `sanity_check`?

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -33,6 +33,23 @@ pipeline {
 
                 } // bandit security check tests
 
+                stage('Safety Check') {
+
+                    steps {
+                        sh 'tox -e safety'
+                    }
+
+                } // safety check
```
Comment (DavidMinarsch) [inline Jenkinsfile]: @marcofavorito why is it commented?

File: Pipfile [hunk 2]
```diff
@@ -33,6 +33,7 @@ tensorflow = "==2.1.0"
 tox = "==3.7.0"
 tox-pipenv = "==1.9.0"
 safety = "==1.8.5"
+liccheck = "*"
```
Comment (DavidMinarsch) [inline Pipfile]: should specify the version

File: strategy.ini [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# some useful links:
+#  https://janelia-flyem.github.io/licenses.html
```
Comment (DavidMinarsch) [inline strategy.ini]: based on this, shouldn't we aim for only MIT, BSD-new, Apache2.0? Do we have any weakly protective or strongly protective ones?

Comment (marcofavorito) [reply strategy.ini]: yes you're right, my bad...

File: strategy.ini [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	; other permissive licenses
+	Historical Permission Notice and Disclaimer (HPND)
+	HPND
+	ISC
+	BSD or Apache License, Version 2.0
+	Modified BSD
+	Expat
+
+unauthorized_licenses:
+;	aliases for MPL 2.0
```
Comment (marcofavorito) [inline strategy.ini]: citing [https://www.mozilla.org/en-US/MPL/2.0/FAQ/](https://www.mozilla.org/en-US/MPL/2.0/FAQ/)

> Q2: Why yet another open source license?
> The MPL fills a useful space in the spectrum of free and open source software licenses, sitting between the Apache license, which does not require modifications to be shared, and the GNU family of licenses, which requires modifications to be shared under a much broader set of circumstances than the MPL.

Hence, MPL is less permissive than Apache 2.0.

File: strategy.ini [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+;	GPL v3
+	GPL v3
+	GPLv3+
+
+[Authorized Packages]
+gym: >=0.15
+;filelock is public domain
+filelock: >=3.0.12
+fetchai-ledger-api: >=0.0.1
```
Comment (marcofavorito) [inline strategy.ini]: See [this issue](https://github.com/fetchai/ledger-api-py/issues/92)

File: strategy.ini [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+chardet: >=3.0.4
+certifi: >=2019.11.28
```
Comment (marcofavorito) [inline strategy.ini]: Although both `certifi` and `chardet` are licensed with a more restrictive license, they are not a first level dependency. Instead, they both are used by requests.

File: examples/protocol_specification_ex/output/two_party_negotiation/__init__.py [hunk 7]
```diff
@@ -1 +1,19 @@
+# -*- coding: utf-8 -*-
```
Comment (DavidMinarsch) [inline examples/protocol_specification_ex/output/two_party_negotiation/__init__.py]: @marcofavorito this file is the output from the generator. It currently does not create a license header. We could do that but then the Copyright must be based on the author specified.

File: examples/protocol_specification_ex/output/two_party_negotiation/message.py [hunk 8]
```diff
@@ -1,3 +1,22 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
```
Comment (DavidMinarsch) [inline examples/protocol_specification_ex/output/two_party_negotiation/message.py]: see above

File: examples/protocol_specification_ex/output/two_party_negotiation/serialization.py [hunk 9]
```diff
@@ -1,3 +1,21 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (DavidMinarsch) [inline examples/protocol_specification_ex/output/two_party_negotiation/serialization.py]: see above

**Pull request merged**